app/controllers/warehouse_reports/cas/rrh_desired_controller.rb

```diff
@@ -9,15 +9,55 @@ class RrhDesiredController < ApplicationController
     include ArelHelper
     include WarehouseReportAuthorization
 
+    before_action :set_filter
+
     def index
-      @clients = client_source.joins(:source_hmis_forms).
-        references(:source_hmis_forms).
-        merge(GrdaWarehouse::HmisForm.rrh_assessment).
-        order(hmis_form_t[:collected_at].desc)
+      @forms = hmis_form_scope.joins(:destination_client).
+        select(*form_columns).
+        order(collected_at: :desc)
+      @forms = @forms.where(collection_location: @collection_location) if @collection_location
+      @forms = @forms.group_by { |f| f.destination_client.id }
+    end
+
+    def set_filter
+      @collection_location = available_locations.detect { |m| params[:forms].try(:[], :location) == m }
+    end
+
+    # NOTE: HmisForm rows tend to be very large, limit the fields we pull
+    private def form_columns
+      [
+        :id,
+        :client_id,
+        :data_source_id,
+        :collection_location,
+        :collected_at,
+        :staff,
+        :staff_email,
+      ] + hmis_form_source.rrh_columns
+    end
+
+    private def client_scope
+      client_source.destination.no_release_on_file
+    end
+
+    private def client_source
+      GrdaWarehouse::Hud::Client
+    end
+
```

Comment:
Dead code?

Reply:
Yep, took a few iterations to get to the right solution for what to show.  Thanks for catching it.